File: haskell/raindrops/Raindrops.hs
module Raindrops(convert) where

convert :: Int -> String
convert number = sandbox $ pling ++ plang ++ plong
    where
        pling = if number `isDivisibleBy` 3 then "Pling" else ""
        plang = if number `isDivisibleBy` 5 then "Plang" else ""
        plong = if number `isDivisibleBy` 7 then "Plong" else ""
        sandbox x = if x /= "" then x else show number
        n `isDivisibleBy` q = n `mod` q == 0

File: lua/raindrops/raindrops.lua
return function(number)
    local drops_result = {}
    if number % 3 == 0 then
        table.insert(drops_result, 'Pling')
    end
    if number % 5 == 0 then
        table.insert(drops_result, 'Plang')
    end
    if number % 7 == 0 then
        table.insert(drops_result, 'Plong')
    end

    if #drops_result > 0 then
        return table.concat(drops_result)
    end

    return tostring(number)
end
